Tests/Integration/ProcessingTests.cs

```diff
@@ -1697,7 +1525,7 @@ public async Task When_Receiving_Device_Message_And_Loading_Device_Fails_Second_
             await Task.Delay(150);
 
             using var request2 = CreateWaitableRequest(simDevice.CreateUnconfirmedDataUpMessage("2"));
-            messageDispatcher.DispatchRequest(request2);
+            messageDispatcher.Value.DispatchRequest(request2);
```

Comment:
Instead of adding .Value all over we could have initialized a stack variable `var messageDispatcher = messageDispatcherWrapper.Value` for example and keep the IDisposable as a "wrapper".

Reply:
Otherwise this is a nice refactoring and helps reducing code and maintainability when changes are made in that area.

Reply:
Done with 47b3e7ed5b61e34b43221a3a37da0ea73a4fc377.